File: Factorial.c
// Factorial of number

#include<stdio.h>

int Factorial(int iValue)
{
    int iFact = 1;
    int iCnt = 0;

    // for(iCnt = 1;iCnt<=iValue;iCnt++)
    // {
    //     iFact = iFact * iCnt;
    // }

    for(iCnt = iValue;iCnt>1;iCnt--)
    {
        iFact = iFact * iCnt;
    }

    return iFact;
}

int main()
{
    int iNo = 0;
    int iRet = 0;

    printf("Enter no.\n");
    scanf("%d",&iNo);

    iRet = Factorial(iNo); 
    printf("Factorial is %d\n",iRet);

    return 0;
}

File: Factorial.java
// FActorial of number

import java.util.Scanner;

public class Factorial 
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number");
        int iValue = sobj.nextInt();

        Number nobj = new Number();

        int iRet = nobj.FactorialX(iValue);

        System.out.println("Factorial is "+iRet);
    }
}

class Number
{
    public int FactorialX(int iNo)
    {
        int iFact = 1;
        // for(int i = iNo; i>0 ;i--)
        // {
        //     iFact = iFact * i;
        // }

        for(int i = 1; i<=iNo ;i++)
        {
            iFact = iFact * i;
        }

        return iFact;
    }
}